clients/src/test/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManagerTest.java

```diff
@@ -1066,6 +1114,7 @@ private <K, V> void buildRequestManager(MetricConfig metricConfig,
                 backgroundEventHandler,
                 metricsManager,
                 shareFetchCollector));
+        shareConsumeRequestManager.setCallbackHandlerConfig(true);
```

Comment:
I wonder if you could make the callback config an argument to this method, and then only turn it on for tests which need the callback. It's a bit odd having the default for testing being true, even though the default for run-time is false.

Reply:
Yeah I was thinking to do that, few tests required the config to be true. Now I have only set the config to true in the required tests and false by default.